\documentclass[]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \ifxetex
    \usepackage{mathspec}
  \else
    \usepackage{fontspec}
  \fi
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}


\usepackage{longtable,booktabs}
\usepackage{graphicx}
% grffile has become a legacy package: https://ctan.org/pkg/grffile
\IfFileExists{grffile.sty}{%
\usepackage{grffile}
}{}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{5}

%%% Use protect on footnotes to avoid problems with footnotes in titles
\let\rmarkdownfootnote\footnote%
\def\footnote{\protect\rmarkdownfootnote}

%%% Change title format to be more compact
\usepackage{titling}

% Create subtitle command for use in maketitle
\providecommand{\subtitle}[1]{
  \posttitle{
    \begin{center}\large#1\end{center}
    }
}

\setlength{\droptitle}{-2em}

\RequirePackage[]{C:/Program Files/R/R-4.3.1/library/BiocStyle/resources/tex/Bioconductor}

\bioctitle[]{The lute user's guide}
    \pretitle{\vspace{\droptitle}\centering\huge}
  \posttitle{\par}
\author{Sean K. Maden and Stephanie C. Hicks}
    \preauthor{\centering\large\emph}
  \postauthor{\par}
      \predate{\centering\large\emph}
  \postdate{\par}
    \date{17 October, 2023}

\newlength{\cslhangindent}
\setlength{\cslhangindent}{1.5em}
% For Pandoc 2.8 to 2.11
\newenvironment{cslreferences}%
  {}%
  {\par}
% For pandoc 2.11+ using new --citeproc
\newlength{\csllabelwidth}
\setlength{\csllabelwidth}{3em}
\newenvironment{CSLReferences}[2] % #1 hanging-ident, #2 entry spacing
 {% don't indent paragraphs
  \setlength{\parindent}{0pt}
  % turn on hanging indent if param 1 is 1
  \ifodd #1 \everypar{\setlength{\hangindent}{\cslhangindent}}\ignorespaces\fi
  % set entry spacing
  \ifnum #2 > 0
  \setlength{\parskip}{#2\baselineskip}
  \fi
 }%
 {}
\usepackage{calc}
\newcommand{\CSLBlock}[1]{#1\hfill\break}
\newcommand{\CSLLeftMargin}[1]{\parbox[t]{\csllabelwidth}{#1}}
\newcommand{\CSLRightInline}[1]{\parbox[t]{\linewidth - \csllabelwidth}{#1}}
\newcommand{\CSLIndent}[1]{\hspace{\cslhangindent}#1}

% code highlighting
\definecolor{fgcolor}{rgb}{0.251, 0.251, 0.251}
\newcommand{\hlnum}[1]{\textcolor[rgb]{0.816,0.125,0.439}{#1}}%
\newcommand{\hlstr}[1]{\textcolor[rgb]{0.251,0.627,0.251}{#1}}%
\newcommand{\hlcom}[1]{\textcolor[rgb]{0.502,0.502,0.502}{\textit{#1}}}%
\newcommand{\hlopt}[1]{\textcolor[rgb]{0,0,0}{#1}}%
\newcommand{\hlstd}[1]{\textcolor[rgb]{0.251,0.251,0.251}{#1}}%
\newcommand{\hlkwa}[1]{\textcolor[rgb]{0.125,0.125,0.941}{#1}}%
\newcommand{\hlkwb}[1]{\textcolor[rgb]{0,0,0}{#1}}%
\newcommand{\hlkwc}[1]{\textcolor[rgb]{0.251,0.251,0.251}{#1}}%
\newcommand{\hlkwd}[1]{\textcolor[rgb]{0.878,0.439,0.125}{#1}}%
\let\hlipl\hlkwb
%
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
%
\newenvironment{Shaded}{\begin{myshaded}}{\end{myshaded}}
% set background for result chunks
\let\oldverbatim\verbatim
\renewenvironment{verbatim}{\color{codecolor}\begin{myshaded}\begin{oldverbatim}}{\end{oldverbatim}\end{myshaded}}
%
\newcommand{\KeywordTok}[1]{\hlkwd{#1}}
\newcommand{\DataTypeTok}[1]{\hlkwc{#1}}
\newcommand{\DecValTok}[1]{\hlnum{#1}}
\newcommand{\BaseNTok}[1]{\hlnum{#1}}
\newcommand{\FloatTok}[1]{\hlnum{#1}}
\newcommand{\ConstantTok}[1]{\hlnum{#1}}
\newcommand{\CharTok}[1]{\hlstr{#1}}
\newcommand{\SpecialCharTok}[1]{\hlstr{#1}}
\newcommand{\StringTok}[1]{\hlstr{#1}}
\newcommand{\VerbatimStringTok}[1]{\hlstr{#1}}
\newcommand{\SpecialStringTok}[1]{\hlstr{#1}}
\newcommand{\ImportTok}[1]{{#1}}
\newcommand{\CommentTok}[1]{\hlcom{#1}}
\newcommand{\DocumentationTok}[1]{\hlcom{#1}}
\newcommand{\AnnotationTok}[1]{\hlcom{#1}}
\newcommand{\CommentVarTok}[1]{\hlcom{#1}}
\newcommand{\OtherTok}[1]{{#1}}
\newcommand{\FunctionTok}[1]{\hlstd{#1}}
\newcommand{\VariableTok}[1]{\hlstd{#1}}
\newcommand{\ControlFlowTok}[1]{\hlkwd{#1}}
\newcommand{\OperatorTok}[1]{\hlopt{#1}}
\newcommand{\BuiltInTok}[1]{{#1}}
\newcommand{\ExtensionTok}[1]{{#1}}
\newcommand{\PreprocessorTok}[1]{\textit{#1}}
\newcommand{\AttributeTok}[1]{{#1}}
\newcommand{\RegionMarkerTok}[1]{{#1}}
\newcommand{\InformationTok}[1]{\textcolor{messagecolor}{#1}}
\newcommand{\WarningTok}[1]{\textcolor{warningcolor}{#1}}
\newcommand{\AlertTok}[1]{\textcolor{errorcolor}{#1}}
\newcommand{\ErrorTok}[1]{\textcolor{errorcolor}{#1}}
\newcommand{\NormalTok}[1]{\hlstd{#1}}
%
\AtBeginDocument{\bibliographystyle{C:/Program Files/R/R-4.3.1/library/BiocStyle/resources/tex/unsrturl}}


\begin{document}
\maketitle

\packageVersion{lute 0.99.6}

\hypertarget{how-to-use-this-guide}{%
\section{How to use this guide}\label{how-to-use-this-guide}}

This User's Guide describes how to use \texttt{lute}, a framework for bulk
transcriptomics deconvolution. An introduction to deconvolution experiments is
provided, followed by a description of the framework and a small experiment
using example data.

Begin by passing the following to your active R session:

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(lute)}
\end{Highlighting}
\end{Shaded}

\hypertarget{deconvolution-overview}{%
\section{Deconvolution overview}\label{deconvolution-overview}}

\hypertarget{definition}{%
\subsection{Definition}\label{definition}}

Deconvolution is the task of quantifying signal from a signal mixture, which
is sometimes called a signal convolution. Deconvolution experiments try to
predict signals from their mixtures as accurately and reliably as possible.

\hypertarget{transcriptomics-deconvolution}{%
\subsection{Transcriptomics deconvolution}\label{transcriptomics-deconvolution}}

In the field of transcriptomics, deconvolution is often applied to gene
expression datasets in order to predict cell type quantities from cell mixtures.
This technique is applied to bulk tissue specimens of multiple cell types, where
accurate cell type quantifications can improve bias adjustments or allow testing
of new hypotheses.

\hypertarget{experiment-elements}{%
\subsection{Experiment elements}\label{experiment-elements}}

Most deconvolution algorithms predict the type-specific proportions from a
signature matrix and a convoluted signals matrix in the following manner:

\[P \leftarrow [Y, Z]\]
Where we have the following term definitions for bulk transcriptomics deconvolution:

\begin{itemize}
\item
  \(P\) : \(K\)-length vector of predicted proportions for the number of cell types \(K\).
\item
  \(Z\) : Signature expression matrix, with dimensions of \(G\) total rows of
  signature genes by \(K\) total columns of cell types.
\item
  \(Y\) : The convoluted signals expression matrix, with dimensions of \(G\) total
  rows of signature genes by \(J\) total columns of bulk samples.
\end{itemize}

The above terms are shared by reference-based deconvolution algorithms, which
are defined by their requirement of a signature matrix \(Z\) to make predictions.

Some additional important properties to consider are the preparation steps used
to generate the terms \(Z\) and \(Y\). These steps might include data rescaling and
transformation of the expression signals, or certain selection criteria to
arrive at the final set of \(G\) signature genes.

\hypertarget{pseudobulking}{%
\subsection{Pseudobulking}\label{pseudobulking}}

Another important representation of deconvolution is in terms of a pseudobulk
equation. A pseudobulk is a simulated bulk sample generated using either cell-
or type-level reference data. Pseudobulk analysis is a common task for
deconvolution experiments.

We may represent the pseudobulked sample \(Y_{PB}\) in terms of the \(Z\) signature
matrix and \(P\) cell type proportions, defined as above, as well as some
\(K\)-length vector \(S\), which contains cell size estimates used for rescaling
\(Z\). This looks like the following:

\[Y_{PB}=Z * P * S\]

Inclusion of the \(S\) vector of cell size scaling factors is an important
strategy to correct for potential bias due to systematic differences in sizes
between the sizes of predicted cell types. It has been used in deconvolution
studies of a variety of tissues, including brain and blood
(Monaco et al. (\protect\hyperlink{ref-monaco_rna-seq_2019}{2019}), Racle and Gfeller (\protect\hyperlink{ref-racle_epic_2020}{2020}), Sosina et al. (\protect\hyperlink{ref-sosina_strategies_2021}{2021})).

\hypertarget{deconvolution-algorithms}{%
\subsection{Deconvolution algorithms}\label{deconvolution-algorithms}}

There are dozens of different deconvolution algorithms used in the field of
trascriptomics alone. These may be organized in several ways, such as whether
they incorporate a variance weighting strategy or cell size scale factors.
Another useful way to organize algorithms is by whether they incorporate the
non-negative least squares (NNLS) algorithm in some way. This is a statistical
approach with the added constraint of assumed non-negativity for inputs, which
holds when we consider typical gene expression datasets in the form of counts
or log-normalized counts.

The next section shows a hierarchical class structure for accessing multiple
deconvolution algorithms, and we can think of this hierarchy as yet another
way of organizing and relating different deconvolution algorithms in a useful
and actionable manner.

\hypertarget{the-lute-framework-for-deconvolution}{%
\section{\texorpdfstring{The \texttt{lute} framework for deconvolution}{The lute framework for deconvolution}}\label{the-lute-framework-for-deconvolution}}

\texttt{lute} supports deconvolution experiments by coupling convenient management of
standard experiment tasks with standard mappings of common inputs to their
synonyms in supported algorithms for marker selection and deconvolution.

Accessing the framework is as simple as running the \texttt{lute()} function using
your experiment data (see \texttt{?lute} for details). Runnable examples using the
framework function are provided below.

\hypertarget{supported-deconvolution-algorithms}{%
\subsection{Supported deconvolution algorithms}\label{supported-deconvolution-algorithms}}

To view a table of supported algorithms and their details, pass the following
code to your R session:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{info.table }\OtherTok{\textless{}{-}} \FunctionTok{lute\_supported\_deconvolution\_algorithms}\NormalTok{()[,}\FunctionTok{seq}\NormalTok{(}\DecValTok{5}\NormalTok{)]}
\NormalTok{knitr}\SpecialCharTok{::}\FunctionTok{kable}\NormalTok{(info.table)}
\end{Highlighting}
\end{Shaded}

\begin{tabular}{l|l|l|l|l}
\hline
method\_shortname & method\_fullname & strict\_method\_used & package & package\_url\\
\hline
nnls & non-negative least squares & nnls & nnls & https://cran.r-project.org/web/packages/nnls/index.html\\
\hline
music & MUlti-Subject SIngle Cell deconvolution & nnls & MuSiC & https://github.com/xuranw/MuSiC\\
\hline
music2 & MUlti-Subject SIngle Cell deconvolution 2 & nnls & MuSiC & https://github.com/xuranw/MuSiC\\
\hline
music2 & MUlti-Subject SIngle Cell deconvolution 2 & nnls & MuSiC2 & https://github.com/Jiaxin-Fan/MuSiC2\\
\hline
EPIC & EPIC &  & EPIC & https://github.com/GfellerLab/EPIC\\
\hline
DeconRNASeq & DeconRNASeq &  & DeconRNASeq & https://bioconductor.org/packages/release/bioc/html/DeconRNASeq.html\\
\hline
Bisque & Bisque & nnls & BisqueRNA & https://github.com/cozygene/bisque\\
\hline
SCDC &  & nnls & SCDC & https://github.com/meichendong/SCDC\\
\hline
\end{tabular}

\hypertarget{the-deconvolutionparam-class-hierarchy}{%
\subsection{\texorpdfstring{The \texttt{deconvolutionParam} class hierarchy}{The deconvolutionParam class hierarchy}}\label{the-deconvolutionparam-class-hierarchy}}

Briefly, \texttt{lute} defines a new class hierarchy that organizes deconvolution
functions. This was based on the approach uesd by the \texttt{bluster} R/Bioconductor
package for clustering algorithms.

Classes in this hierarchy each have a method defined for the \texttt{deconvolution()}
generic function (see \texttt{?deconvolution} for details). Methods for higher-level
parent classes like \texttt{referencebasedParam} will perform data summary and marker
genes comparisons. For the algorithm-specific classes, such as \texttt{nnlsParam} or
\texttt{musicParam}, the method maps standard inputs like \(Y\), \(Z\), and \(S\) to their
algorithm-specific synonyms.

The \texttt{deconvolutionParam} class hierarchy for supported algorithms is visualized
in the following diagram:

The \texttt{deconvolutionParam} class hierarchy is intended to be extensible to new
algorithms. A future vignette will provide a step-by-step guide to supporting
a new algorithms using these classes and their methods.

\hypertarget{deconvolution-experiment-example}{%
\section{Deconvolution experiment example}\label{deconvolution-experiment-example}}

This section features several runnable examples of deconvolution experiments
using the \texttt{lute()} framework function.

We may begin with an object of type \texttt{SingleCellExperiment} (a.k.a. ``sce'' object)
containing cell-level expression data. We may use the main \texttt{lute()} framework
function to perform a deconvolution experiment with these data.

For demonstration, we generate a randomized sce object with \texttt{random\_sce()} (see
\texttt{?random\_sce} for details). Set the \(G\) marker genes to be 10, or 5 per cell
type, which we will identify from an initial set of 100 genes:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{markers.per.type }\OtherTok{\textless{}{-}} \DecValTok{5}
\NormalTok{total.genes }\OtherTok{\textless{}{-}} \DecValTok{100}
\NormalTok{sce.example }\OtherTok{\textless{}{-}} \FunctionTok{random\_sce}\NormalTok{(}\AttributeTok{num.genes=}\NormalTok{total.genes)}
\end{Highlighting}
\end{Shaded}

We could identify marker genes in \texttt{sce.example} now by running \texttt{lute} without
setting the deconvolution algorithm argument:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{markers }\OtherTok{\textless{}{-}} \FunctionTok{lute}\NormalTok{(}\AttributeTok{sce=}\NormalTok{sce.example, }\AttributeTok{markers.per.type=}\NormalTok{markers.per.type, }
                \AttributeTok{deconvolution.algorithm=}\ConstantTok{NULL}\NormalTok{)}
\DocumentationTok{\#\# Parsing marker gene arguments...}
\DocumentationTok{\#\# Using meanratiosParam...}
\DocumentationTok{\#\# selecting among 100 genes for markers of type: type1...}
\DocumentationTok{\#\# Selecting by gene}
\DocumentationTok{\#\# selecting among 95 genes for markers of type: type2...}
\DocumentationTok{\#\# Selecting by gene}
\DocumentationTok{\#\# Filtering sce...}
\FunctionTok{length}\NormalTok{(markers}\SpecialCharTok{$}\NormalTok{typemarker.results)}
\DocumentationTok{\#\# [1] 10}
\end{Highlighting}
\end{Shaded}

We identified 10 marker genes from the provided data.

To run a more complete experiment, we need to define \(Y\). For this demonstration,
we define \(Y\) as a pseudobulk from the provided \texttt{sce.example} data using
\texttt{ypb\_from\_sce()} (see \texttt{?ypb\_from\_sce} for details).

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{input\_y }\OtherTok{\textless{}{-}} \FunctionTok{ypb\_from\_sce}\NormalTok{(}\AttributeTok{sce=}\NormalTok{sce.example, }
                  \AttributeTok{assay.name=}\StringTok{"counts"}\NormalTok{, }
                  \AttributeTok{celltype.variable=}\StringTok{"celltype"}\NormalTok{)}
\NormalTok{input\_y }\OtherTok{\textless{}{-}} \FunctionTok{as.matrix}\NormalTok{(input\_y)}
\end{Highlighting}
\end{Shaded}

By default, the pseudobulk used all available cell data in \texttt{sce.example}.

Finally, we run marker selection and deconvolution in succession with \texttt{lute} as
follows:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{experiment.results }\OtherTok{\textless{}{-}} \FunctionTok{lute}\NormalTok{(}\AttributeTok{sce=}\NormalTok{sce.example, }\AttributeTok{y=}\NormalTok{input\_y, }
                           \AttributeTok{typemarker.algorithm=}\ConstantTok{NULL}\NormalTok{)}
\DocumentationTok{\#\# Parsing deconvolution arguments...}
\DocumentationTok{\#\# Using NNLS...}
\end{Highlighting}
\end{Shaded}

By default, \texttt{lute} used the mean ratios algorithm to get markers, then the
NNLS algorithm to get cell type proportion predictions.

We can inspect the predicted cell type proportions \(P\) from \texttt{experiment.results}:

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{experiment.results}\SpecialCharTok{$}\NormalTok{deconvolution.results}
\DocumentationTok{\#\# Number of bulk samples (J): 1}
\DocumentationTok{\#\# Number of cell types (K): 2}
\DocumentationTok{\#\# Cell type labels:}
\DocumentationTok{\#\#  type1;  type2}
\DocumentationTok{\#\# predictions.table summary:}
\DocumentationTok{\#\# NULL}
\DocumentationTok{\#\#   type1 type2}
\DocumentationTok{\#\# 1   0.5   0.5}
\end{Highlighting}
\end{Shaded}

\hypertarget{conclusions-and-further-reading}{%
\section{Conclusions and further reading}\label{conclusions-and-further-reading}}

This User's Guide introduced the \texttt{lute} framework for deconvolution, including
an outline of deconvolution experiment variables, an introduction to the
different deconvolution methods for bulk transcriptomics, and a small runnable
example showing how to access the NNLS algorithm by calling the \texttt{deconvolution}
generic on an object of class \texttt{nnlsParam}. If you found this guide useful, you
may also find the other vignettes included in \texttt{lute} to be helpful.

\hypertarget{session-info}{%
\section{Session Info}\label{session-info}}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{sessionInfo}\NormalTok{()}
\DocumentationTok{\#\# R version 4.3.1 (2023{-}06{-}16 ucrt)}
\DocumentationTok{\#\# Platform: x86\_64{-}w64{-}mingw32/x64 (64{-}bit)}
\DocumentationTok{\#\# Running under: Windows 11 x64 (build 22621)}
\DocumentationTok{\#\# }
\DocumentationTok{\#\# Matrix products: default}
\DocumentationTok{\#\# }
\DocumentationTok{\#\# }
\DocumentationTok{\#\# locale:}
\DocumentationTok{\#\# [1] LC\_COLLATE=English\_United States.utf8 }
\DocumentationTok{\#\# [2] LC\_CTYPE=English\_United States.utf8   }
\DocumentationTok{\#\# [3] LC\_MONETARY=English\_United States.utf8}
\DocumentationTok{\#\# [4] LC\_NUMERIC=C                          }
\DocumentationTok{\#\# [5] LC\_TIME=English\_United States.utf8    }
\DocumentationTok{\#\# }
\DocumentationTok{\#\# time zone: America/Los\_Angeles}
\DocumentationTok{\#\# tzcode source: internal}
\DocumentationTok{\#\# }
\DocumentationTok{\#\# attached base packages:}
\DocumentationTok{\#\# [1] stats4    stats     graphics  grDevices utils     datasets  methods  }
\DocumentationTok{\#\# [8] base     }
\DocumentationTok{\#\# }
\DocumentationTok{\#\# other attached packages:}
\DocumentationTok{\#\#  [1] lute\_0.99.6                 SingleCellExperiment\_1.22.0}
\DocumentationTok{\#\#  [3] SummarizedExperiment\_1.30.2 Biobase\_2.60.0             }
\DocumentationTok{\#\#  [5] GenomicRanges\_1.52.0        GenomeInfoDb\_1.36.1        }
\DocumentationTok{\#\#  [7] IRanges\_2.34.1              S4Vectors\_0.38.1           }
\DocumentationTok{\#\#  [9] BiocGenerics\_0.46.0         MatrixGenerics\_1.12.2      }
\DocumentationTok{\#\# [11] matrixStats\_1.0.0           BiocStyle\_2.28.0           }
\DocumentationTok{\#\# }
\DocumentationTok{\#\# loaded via a namespace (and not attached):}
\DocumentationTok{\#\#  [1] xfun\_0.39                 lattice\_0.21{-}8           }
\DocumentationTok{\#\#  [3] vctrs\_0.6.3               tools\_4.3.1              }
\DocumentationTok{\#\#  [5] bitops\_1.0{-}7              generics\_0.1.3           }
\DocumentationTok{\#\#  [7] parallel\_4.3.1            tibble\_3.2.1             }
\DocumentationTok{\#\#  [9] fansi\_1.0.4               cluster\_2.1.4            }
\DocumentationTok{\#\# [11] pkgconfig\_2.0.3           BiocNeighbors\_1.18.0     }
\DocumentationTok{\#\# [13] Matrix\_1.6{-}0              nnls\_1.4                 }
\DocumentationTok{\#\# [15] sparseMatrixStats\_1.12.2  dqrng\_0.3.0              }
\DocumentationTok{\#\# [17] lifecycle\_1.0.3           GenomeInfoDbData\_1.2.10  }
\DocumentationTok{\#\# [19] compiler\_4.3.1            statmod\_1.5.0            }
\DocumentationTok{\#\# [21] bluster\_1.10.0            codetools\_0.2{-}19         }
\DocumentationTok{\#\# [23] htmltools\_0.5.5           RCurl\_1.98{-}1.12          }
\DocumentationTok{\#\# [25] yaml\_2.3.7                pillar\_1.9.0             }
\DocumentationTok{\#\# [27] crayon\_1.5.2              BiocParallel\_1.34.2      }
\DocumentationTok{\#\# [29] limma\_3.56.2              DelayedArray\_0.26.6      }
\DocumentationTok{\#\# [31] metapod\_1.8.0             locfit\_1.5{-}9.8           }
\DocumentationTok{\#\# [33] tidyselect\_1.2.0          rsvd\_1.0.5               }
\DocumentationTok{\#\# [35] digest\_0.6.33             BiocSingular\_1.16.0      }
\DocumentationTok{\#\# [37] dplyr\_1.1.2               bookdown\_0.34            }
\DocumentationTok{\#\# [39] fastmap\_1.1.1             grid\_4.3.1               }
\DocumentationTok{\#\# [41] cli\_3.6.1                 magrittr\_2.0.3           }
\DocumentationTok{\#\# [43] S4Arrays\_1.0.4            utf8\_1.2.3               }
\DocumentationTok{\#\# [45] edgeR\_3.42.4              DelayedMatrixStats\_1.22.1}
\DocumentationTok{\#\# [47] rmarkdown\_2.23            XVector\_0.40.0           }
\DocumentationTok{\#\# [49] igraph\_1.5.0              scran\_1.28.1             }
\DocumentationTok{\#\# [51] ScaledMatrix\_1.8.1        beachmat\_2.16.0          }
\DocumentationTok{\#\# [53] evaluate\_0.21             knitr\_1.43               }
\DocumentationTok{\#\# [55] irlba\_2.3.5.1             rlang\_1.1.1              }
\DocumentationTok{\#\# [57] Rcpp\_1.0.11               scuttle\_1.10.1           }
\DocumentationTok{\#\# [59] glue\_1.6.2                BiocManager\_1.30.21.1    }
\DocumentationTok{\#\# [61] rstudioapi\_0.15.0         R6\_2.5.1                 }
\DocumentationTok{\#\# [63] zlibbioc\_1.46.0}
\end{Highlighting}
\end{Shaded}

\hypertarget{works-cited}{%
\section*{Works cited}\label{works-cited}}
\addcontentsline{toc}{section}{Works cited}

\hypertarget{refs}{}
\begin{CSLReferences}{1}{0}
\leavevmode\vadjust pre{\hypertarget{ref-monaco_rna-seq_2019}{}}%
Monaco, Gianni, Bernett Lee, Weili Xu, Seri Mustafah, You Yi Hwang, Christophe Carr√©, Nicolas Burdin, et al. 2019. {``{RNA}-{Seq} {Signatures} {Normalized} by {mRNA} {Abundance} {Allow} {Absolute} {Deconvolution} of {Human} {Immune} {Cell} {Types}.''} \emph{Cell Reports} 26 (6): 1627--1640.e7. \url{https://doi.org/10.1016/j.celrep.2019.01.041}.

\leavevmode\vadjust pre{\hypertarget{ref-racle_epic_2020}{}}%
Racle, Julien, and David Gfeller. 2020. {``{EPIC}: {A} {Tool} to {Estimate} the {Proportions} of {Different} {Cell} {Types} from {Bulk} {Gene} {Expression} {Data}.''} Edited by Sebastian Boegel, Methods in {Molecular} {Biology}, 233--48. \url{https://doi.org/10.1007/978-1-0716-0327-7_17}.

\leavevmode\vadjust pre{\hypertarget{ref-sosina_strategies_2021}{}}%
Sosina, Olukayode A., Matthew N. Tran, Kristen R. Maynard, Ran Tao, Margaret A. Taub, Keri Martinowich, Stephen A. Semick, et al. 2021. {``Strategies for Cellular Deconvolution in Human Brain {RNA} Sequencing Data,''} no. 10:750 (August). \url{https://doi.org/10.12688/f1000research.50858.1}.

\end{CSLReferences}


\end{document}
